common/libs/units/units.h

```diff
@@ -115,12 +115,38 @@ template <class Q, class ValTy> class ArithScalar : public Scalar<Q, ValTy> {
 public:
   constexpr Q operator+(const ArithScalar &a) { return Q(this->val_ + a.val_); }
   constexpr Q operator-(const ArithScalar &a) { return Q(this->val_ - a.val_); }
+  inline ArithScalar &operator+=(const ArithScalar &a) {
+    return *this = *this + a;
+  }
+  inline ArithScalar &operator-=(const ArithScalar &a) {
+    return *this = *this - a;
+  }
 
 protected:
   // Pull in base class's constructor.
   using Scalar<Q, ValTy>::Scalar;
 };
 
+// Represents an ArithScalar where multiplication by an external factor of type
+// ValTy is defined, creating what mathematicians call a "module":
+// https://en.wikipedia.org/wiki/Module_(mathematics)
+// Multiplication by a ValTy factor follows the rules of multiplication for
+// ValTy, and requires multiplication to be well defined on ValTy type, except
+// that it does not commute: float*ModuleScalar is undefined
```

Comment:
Any reason not to make it commute?  That seems like it'd be useful.

Reply:
Other than the fact I couldn't figure out how to make it work, no.

Reply:
I think it should be friend functions, like you did with `constexpr friend Volume operator*(const VolumetricFlow &a, const Duration &b);`

Reply:
Got it working, thanks!